hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/CustomSourceFaultToleranceTopologyChangedTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.pipeline;
+
+import com.hazelcast.core.IList;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.aggregate.AggregateOperations;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.core.JobStatus;
+import com.hazelcast.jet.datamodel.WindowResult;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.util.UuidUtil;
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.concurrent.CancellationException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.jet.config.ProcessingGuarantee.EXACTLY_ONCE;
+import static com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually;
+import static com.hazelcast.jet.core.JetTestSupport.assertTrueEventually;
+import static com.hazelcast.jet.pipeline.WindowDefinition.tumbling;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class CustomSourceFaultToleranceTopologyChangedTest extends JetTestSupport {
```

Comment:
it would better to name this `SourceBuilder_TopologyChangeTest`, we use this convention in other places too.

Reply:
It makes sense -> renamed. Thanks.